kbase-extension/static/kbase/js/widgets/appWidgets2/validators/struct.js

```diff
@@ -1,121 +1,103 @@
-                    validationResult.messageId = Constants.MESSAGE_IDS.REQUIRED_MISSING;
-                }
-                return validationResult;
-            } else {
-                const validationMap = {};
-                Object.keys(spec.parameters.specs).forEach((id) => {
-                    const paramSpec = spec.parameters.specs[id];
-                    const paramValue = value[id];
-                    validationMap[id] = ValidationResolver.validate(paramValue, paramSpec);
-                });
-                return Promise.props(validationMap).then((subValidationResults) => {
-                    /*
-                            Propagation rules for struct validation:
-                            optional:
-                            if any subcontrol has an error, so do we, and the message
-                            is just generic (Error in subcontrol).
-                            if any are required and empty, don't do anything (upper context will
-                            catch this)
-                            if all are optional-empty, we are optional-empty
-
-                            required:
-                            if any is a validation error, this becomes error (like above)
-                            otherwise, if any is required-missing, this becomes required-missing
-                            otherwise, if all are optional-empty, we are required-missing
-                        */
-                    let resolved = false;
-                    let subFieldsEmpty = true;
-                    const subParamIds = Object.keys(subValidationResults);
-                    for (const element of subParamIds) {
-                        const result = subValidationResults[element];
-                        if (!result.isValid) {
-                            validationResult.isValid = false;
-                            if (result.diagnosis === Constants.DIAGNOSIS.REQUIRED_MISSING) {
-                                validationResult.diagnosis = Constants.DIAGNOSIS.REQUIRED_MISSING;
-                                validationResult.messageId = Constants.MESSAGE_IDS.REQUIRED_MISSING;
-                            } else {
-                                validationResult.diagnosis = Constants.DIAGNOSIS.INVALID;
-                                validationResult.messageId = 'subfield-invalid';
-                                validationResult.message = 'A sub-field is invalid';
-                            }
-                            resolved = true;
-                            break;
+                    required:
+                    if any is a validation error, this becomes error (like above)
+                    otherwise, if any is required-missing, this becomes required-missing
+                    otherwise, if all are optional-empty, we are required-missing
+                */
+                const validationResult = {
```

Comment:
removed a level of nesting and initialised this variable here, as it's not used in the prior code